libs/auth-api-lib/src/lib/delegations/delegations-incoming.service.ts

```diff
@@ -256,6 +256,51 @@ export class DelegationsIncomingService {
     return [...mergedDelegationMap.values()]
   }
 
+  async verifyDelegationAtProvider(
+    user: User,
+    fromNationalId: string,
+    delegationTypes: AuthDelegationType[],
+  ): Promise<boolean> {
+    const providers = await this.delegationProviderService.findProviders(
+      delegationTypes,
+    )
+
+    if (
+      providers.includes(AuthDelegationProvider.DistrictCommissionersRegistry)
+    ) {
+      let delegationFound = false
+      let isError = false
+      try {
+        delegationFound = await this.syslumennService.checkIfDelegationExists(
+          user.nationalId,
+          fromNationalId,
+        )
+      } catch (error) {
+        isError = true
+        logger.error(
+          `Failed checking if delegation exists at provider '${AuthDelegationProvider.DistrictCommissionersRegistry}'`,
+        )
+      }
+
+      if (delegationFound) {
+        return true
+      } else if (!isError) {
+        Promise.all(
+          delegationTypes.map((dt) =>
+            this.delegationsIndexService.removeDelegationRecord({
+              fromNationalId,
+              toNationalId: user.nationalId,
+              type: dt,
+              provider: AuthDelegationProvider.DistrictCommissionersRegistry,
+            }),
+          ),
+        )
+      }
```

Comment:
You should be able to simplify this similar to my other suggestion.

Reply:
Done.